TypeScript error: Cannot find name 'MessageFunctionInternal'

### Reporting a bug?

When using Vue I18n 9 with Vuetify and TypeScript, the following error message is output after running `npm run build`:
```
node_modules/@intlify/core-base/dist/core-base.d.ts:645:77 - error TS2304: Cannot find name 'MessageFunctionInternal'.

645 export declare type MessageFunction<T = string> = MessageFunctionCallable | MessageFunctionInternal<T>;
                                                                                ~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error in node_modules/@intlify/core-base/dist/core-base.d.ts:645
```

The problem doesn't occur when using `npm run dev`.

### Expected behavior

A project using Vuetify and Vue I18n should build successfully.

### Reproduction

Run:
```sh
npm create vuetify
# Project name: ... vuetify-project
# Which preset would you like to install? » Essentials (Vuetify, VueRouter, Pinia)
# Use TypeScript? ... Yes
# Would you like to install dependencies with yarn, npm, or pnpm? » npm
cd ./vuetify-project
npm install vue-i18n@9
```
then add 
```ts
import { createI18n } from 'vue-i18n'
```
to `src/plugins/index.ts` and run `npm run build`.

Link to reproduction repo: https://stackblitz.com/edit/vitejs-vite-nei8kc?file=src%2Fplugins%2Findex.ts

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (12) x64 AMD Ryzen 5 5500U with Radeon Graphics
    Memory: 1.14 GB / 7.31 GB
  Binaries:
    Node: 18.17.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - C:\Program Files\nodejs\yarn.CMD
    npm: 9.8.1 - C:\Program Files\nodejs\npm.CMD
    pnpm: 8.7.4 - C:\Program Files\nodejs\pnpm.CMD
  Browsers:
    Edge: Spartan (44.22621.2134.0), Chromium (116.0.1938.69)
    Internet Explorer: 11.0.22621.1
  npmPackages:
    @vitejs/plugin-vue: ^4.0.0 => 4.3.4
    @vue/eslint-config-typescript: ^11.0.0 => 11.0.3
    vite: ^4.2.0 => 4.4.9
    vite-plugin-vuetify: ^1.0.0 => 1.0.2
    vue: ^3.2.0 => 3.3.4
    vue-i18n: ^9.3.0 => 9.3.0
    vue-router: ^4.0.0 => 4.2.4
    vue-tsc: ^1.2.0 => 1.8.10
    vuetify: ^3.0.0 => 3.3.15
```


### Screenshot

_No response_

### Additional context

_No response_

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/intlify/vue-i18n-next/blob/master/.github/CONTRIBUTING.md)
- [X] Read the [Documentation](https://vue-i18n.intlify.dev/)
- [X] Check that there isn't [already an issue](https://github.com/intlify/vue-i18n-next/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussions](https://github.com/intlify/vue-i18n-next/discussions)

--- a/packages/core-base/src/context.ts
+++ b/packages/core-base/src/context.ts
@@ -19,7 +19,9 @@ import { resolveWithKeyValue } from './resolver'
 import { fallbackWithSimple } from './fallbacker'
 
 import type { CompileOptions, ResourceNode } from '@intlify/message-compiler'
-import type { VueDevToolsEmitter } from '@intlify/vue-devtools'
+// HACK: embbed `@intlify/vue-devtools` types to `.d.ts`, because `@intlify/dev-tools` is devDependencies
+// TODO: Consider this type dependency when separating into intlify/core
+import type { VueDevToolsEmitter } from '../../vue-devtools/src/index'
 import type { Path, MessageResolver } from './resolver'
 import type {
   Locale,

--- a/packages/core-base/src/devtools.ts
+++ b/packages/core-base/src/devtools.ts
@@ -1,9 +1,10 @@
-import { IntlifyDevToolsHooks } from '@intlify/devtools-if'
-
-import type {
+// HACK: embbed `@intlify/devtools-if` enum and type to `.d.ts`, because `@intlify/dev-tools` is devDependencies
+// TODO: Consider this type dependency when separating into intlify/core
+import {
+  IntlifyDevToolsHooks,
   IntlifyDevToolsEmitter,
   IntlifyDevToolsHookPayloads
-} from '@intlify/devtools-if'
+} from '../../devtools-if/src/index'
 
 let devtools: IntlifyDevToolsEmitter | null = null
 

--- a/packages/core-base/src/runtime.ts
+++ b/packages/core-base/src/runtime.ts
@@ -46,7 +46,6 @@ export type MessageFunctionCallable = <T = string>(
   ctx: MessageContext<T>
 ) => MessageFunctionReturn<T>
 
-/** * @internal */
 export type MessageFunctionInternal<T = string> = {
   (ctx: MessageContext<T>): MessageFunctionReturn<T>
   key?: string

